packages/story-editor/src/app/media/utils/getResourceFromLocalFile.js

```diff
@@ -76,17 +75,25 @@ const getVideoResource = async (file) => {
   const alt = getFileName(file);
   const mimeType = file.type;
 
+  let length = 0;
+  let lengthFormatted = '';
+
   const reader = await createFileReader(file);
 
   const src = createBlob(new Blob([reader.result], { type: mimeType }));
 
   // Here we are potentially dealing with an unsupported file type (e.g. MOV)
   // that cannot be *played* by the browser, but could still be used for generating a poster.
 
-  const videoEl = await preloadVideo(src);
-  const canPlayVideo = '' !== videoEl.canPlayType(mimeType);
+  const videoEl = document.createElement('video');
+  videoEl.src = src;
+  videoEl.addEventListener('loadedmetadata', () => {
+    const lengthData = getVideoLength(videoEl);
+    length = lengthData.length;
+    lengthFormatted = lengthData.lengthFormatted;
+  });
```

Comment:
This won't work because it's not in a promise. There's gonna be race conditions.

Hence I'd highly recommend using `preloadVideoMetaData` instead.

That said, I think this uses `preloadVideo` for a reason.

Need to ensure that the length information and the `canPlayVideo` value, as well as the `hasAudio` stuff below still works as expected.

Reply:
But it was work.. Why does it have to be a promise here? We are loading a resource here, it is a source url. Why did it work before ( before it was a promise ) and why does it work?

Reply:
Then it was wrong before. It won't work **_reliably_** due to a possible race condition. If the `loadedmetadata` event fires only after 10 seconds the whole code below will have been executed already. Hence the need for `preloadVideoMetadata`